datasources/src/main/java/org/opensearch/sql/datasources/service/DataSourceServiceImpl.java

```diff
@@ -90,14 +86,27 @@ public void createDataSource(DataSourceMetadata metadata) {
   public void updateDataSource(DataSourceMetadata dataSourceMetadata) {
     validateDataSourceMetaData(dataSourceMetadata);
     if (!dataSourceMetadata.getName().equals(DEFAULT_DATASOURCE_NAME)) {
-      this.dataSourceLoaderCache.getOrLoadDataSource(dataSourceMetadata);
       this.dataSourceMetadataStorage.updateDataSourceMetadata(dataSourceMetadata);
     } else {
       throw new UnsupportedOperationException(
           "Not allowed to update default datasource :" + DEFAULT_DATASOURCE_NAME);
     }
   }
 
+  @Override
+  public void patchDataSource(Map<String, Object> dataSourceData) {
+    if (!dataSourceData.get(NAME_FIELD).equals(DEFAULT_DATASOURCE_NAME)) {
+      DataSourceMetadata dataSourceMetadata =
+          getRawDataSourceMetadata((String) dataSourceData.get(NAME_FIELD));
+      replaceOldDatasourceMetadata(dataSourceData, dataSourceMetadata);
+      updateDataSource(dataSourceMetadata);
+      this.dataSourceMetadataStorage.patchDataSourceMetadata(dataSourceData);
```

Comment:
Do we require this?
updateDataSource(dataSourceMetadata);
This method already contains this right?